File: react-native-pytorch-core/ios/Canvas/CanvasManager.m
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_REMAP_MODULE(PyTorchCoreCanvasView, CanvasManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onContext2D, RCTBubblingEventBlock)
RCT_EXTERN_METHOD(requiresMainQueueSetup)
@end

File: react-native-pytorch-core/ios/Canvas/CanvasManager.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

@objc(CanvasManager)
class CanvasManager: RCTViewManager {

    override func view() -> UIView! {
        let canvas = DrawingCanvasView()
        canvas.backgroundColor = .white
        return canvas
    }

    override static func requiresMainQueueSetup() -> Bool {
       return true
     }
}
